tests/test_server_proto.py

```diff
@@ -2367,6 +2341,31 @@ async def test_server_proto_tx_31(self):
             await self.con.query('SELECT 42'),
             [42])
 
+    async def test_server_proto_tx_32(self):
+        # Test state sync across 2 frontend connections works fine
+        con2 = await self.connect(database=self.con.dbname)
+        try:
+            await con2.query('''
+                CONFIGURE SESSION
+                    SET default_transaction_isolation := 'RepeatableRead';
+            ''')
+            await self.assert_tx_isolation_and_default(
+                'RepeatableRead', conn=con2
+            )
+
+            # Try a few times back and forth - this should be enough to hit
+            # the same backend connection.
```

Comment:
Of course, that being true indicates that maybe we should do a little more work to try to select connections with matching state more consistently?

(Not now, though. And unfortunately doing that will make these bugs much subtler, too.)

Reply:
Yeah, the connection pool could also be smarter to open new backend connections on state mismatch when the load is low. (cc @mmastrac)